### ISSUE TITLE
AccessViolationException crash in clrjit.dll when running System.Numerics.Vector<T> SIMD code (repro included)

### ISSUE THREAD

**Issue by Jorenkv** [dotnet/coreclr#1407]
I was playing around with the register-sized SIMD operations support provided by System.Numerics.Vector<T> and stumbled across this crash in RyuJIT. Here's a minimal repro:

**Note: only reproduces for a Release build targeting x64 with optimizations enabled. Of course you can't attach a debugger and have 'Suppress JIT optimization on module load' enabled either.**

```csharp
using System.Numerics;

namespace SIMDBugRepro
{
    class Program
    {
        // This code causes the following exception to appear when running a Release build targeting x64 with optimizations enabled.
        //
        // Exception thrown at 0x000007FEF0282427 (clrjit.dll) in SIMDBugRepro.exe: 0xC0000005: Access violation writing location 0x0000000000000030.
        //
        static void Main(string[] args)
        {
            var x = new Vector<float>();
            var y = new Vector<float>();

            // Note that simply doing y += x; does not cause the crash to appear
            float a = 0;
            y += a * x;

            // Uncomment these lines and the crash disappears.
            // In fact adding any operation that 'publishes' y makes the crash go away.
            // Thankfully this makes for an easy workaround - do anything useful with the computation.
            //float[] array = new float[20];
            //y.CopyTo(array, 0);
        }
    }
}
```

Note that the exception does *not* happen when `Main` runs, but rather when `Main` is jitted. To show this, make the code unreachable (but in a way that the compiler can't prove that the code is unreachable so that it doesn't remove the dead code altogether). E.g. insert

```csharp
            if (args != null)
                return;
```

as the first two lines of `Main`. The crash still occurs, even though the relevant code never executes.

**Comment by mikedn** [comment]
This happens with the released .NET 4.6 but doesn't happen with CoreCLR built from current sources. Looks like it was already fixed by #1293

**Comment by cmckinsey** [comment]
@CarolEidt PTAL to confirm.

**Comment by CarolEidt** [comment]
Yes, this was fixed by #1293 but has not yet made it into the released version. As @Jorenkv notes, there is a workaround - this bug will only manifest if the result of a Vector computation is unused.

**Comment by Jorenkv** [comment]
Good to hear, thanks for looking into it. I did search the issue tracker but didn't find anything, too bad it doesn't search commit messages within pull requests or I would have found https://github.com/dotnet/coreclr/pull/1293 that way.

I guess I could have at tested on the latest builds on github before posting this. I'd have to get a build running on my machine first though.

Thanks again.
